### PR TITLE
Fix vaadin-component-demo for IE11

### LINKED ISSUE #3: The demos don't open up in IE11

### PR DESCRIPTION
**Pull request by tomivirkki**
Fixes #3

---
This change is [Reviewable](https://reviewable.io/reviews/vaadin/vaadin-demo-helpers/7)

### CODE REVIEW

File: vaadin-component-demo.html [hunk 1]
```diff
@@ -145,9 +145,10 @@
 
       _configChanged() {
         if (this.configSrc) {
-          fetch(this.configSrc)
-            .then(res => res.json())
-            .then(json => {
+          var xhr = new XMLHttpRequest();
+          xhr.onreadystatechange = () => {
```
Comment (limonte) [inline vaadin-component-demo.html]: `onload` is more readable, you won't need to check `readyState`, just `status` http://youmightnotneedjquery.com/

**Comment by manolo** [comment]
Review status: 0 of 1 files reviewed at latest revision, 4 unresolved discussions.

---

*[vaadin-component-demo.html, line 150 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-demo-helpers/7#-KwpOI2FUMwWTQ01MDLo:-KwpOI2FUMwWTQ01MDLp:b-g52lq6) ([raw file](https://github.com/vaadin/vaadin-demo-helpers/blob/6558823a096cf87e9617403dcc76540254e91907/vaadin-component-demo.html#L150)):*
> ```HTML
>           var xhr = new XMLHttpRequest();
>           xhr.onreadystatechange = () => {
>             if (xhr.readyState == XMLHttpRequest.DONE && xhr.status == 200) {
> ```

I would take advantage of this PR and introduce a console warning if the xhr fails. Not blocker

---

*[vaadin-component-demo.html, line 163 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-demo-helpers/7#-KwpNwo34KNq_GCc8uDf:-KwpNwo34KNq_GCc8uDg:b-n2vg63) ([raw file](https://github.com/vaadin/vaadin-demo-helpers/blob/6558823a096cf87e9617403dcc76540254e91907/vaadin-component-demo.html#L163)):*
> ```HTML
>             }
>           }
>           xhr.open("GET", this.configSrc, true);
> ```

default is `true`, this parameter is not used anymore since `false` (sync) was deprecated a long ago.

---

*[vaadin-component-demo.html, line 171 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-demo-helpers/7#-KwpOgMZrv2JDJkTIxo9:-KwpOgMZrv2JDJkTIxoA:bjkejml) ([raw file](https://github.com/vaadin/vaadin-demo-helpers/blob/6558823a096cf87e9617403dcc76540254e91907/vaadin-component-demo.html#L171)):*
> ```HTML
>         if (!selectedPage || !this._demoConfig) return;
> 
>         let page;
> ```

This could be more concise just using tertiary `let page = selectedPage ? .... : ...`. Not blocker

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-demo-helpers/7)*

**Comment by manolo** [comment]
Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-demo-helpers/7)*

**Comment by tomivirkki** [comment]
Review status: 0 of 1 files reviewed at latest revision, 4 unresolved discussions.

---

*[vaadin-component-demo.html, line 150 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-demo-helpers/7#-KwpOI2FUMwWTQ01MDLo:-KwpcnytGcpdPA8gutLx:b-tm8wi0) ([raw file](https://github.com/vaadin/vaadin-demo-helpers/blob/6558823a096cf87e9617403dcc76540254e91907/vaadin-component-demo.html#L150)):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary><blockquote>

I would take advantage of this PR and introduce a console warning if the xhr fails. Not blocker
</blockquote></details>

Irrelevant for this PR

---

*[vaadin-component-demo.html, line 163 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-demo-helpers/7#-KwpNwo34KNq_GCc8uDf:-Kwpcx6KZGF3g2PFPj6h:b-896fix) ([raw file](https://github.com/vaadin/vaadin-demo-helpers/blob/6558823a096cf87e9617403dcc76540254e91907/vaadin-component-demo.html#L163)):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary><blockquote>

default is `true`, this parameter is not used anymore since `false` (sync) was deprecated a long ago.
</blockquote></details>

Done.

---

*[vaadin-component-demo.html, line 171 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-demo-helpers/7#-KwpOgMZrv2JDJkTIxo9:-KwpcjL5ckuAYGFlDCFe:b-tm8wi0) ([raw file](https://github.com/vaadin/vaadin-demo-helpers/blob/6558823a096cf87e9617403dcc76540254e91907/vaadin-component-demo.html#L171)):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary><blockquote>

This could be more concise just using tertiary `let page = selectedPage ? .... : ...`. Not blocker
</blockquote></details>

Irrelevant for this PR

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-demo-helpers/7)*

**Comment by manolo** [comment]
Reviewed 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-demo-helpers/7)*

**Comment by limonte** [comment]
Reviewed 1 of 1 files at r3.
Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-demo-helpers/7)*

**Comment by manolo** [comment]
:lgtm:

---

Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-demo-helpers/7#-:-KwsfiQuV6G1MYVdcUL7:bnfp4nl)*

**Pull request merged**